pkg/custom_detectors/CUSTOM_DETECTORS.md

```diff
@@ -0,0 +1,200 @@
+   - Execute TruffleHog, specifying your configuration file:
+
+     ```bash
+     trufflehog filesystem <path_to_folder_or_file> --config=<path_to_file>/config.yaml
+     ```
+
+   - Replace `<path_to_folder_or_file>` with the path to the directory or file you want to scan, and `<path_to_file>` with the path to your `config.yaml`.
+   - TruffleHog will scan the specified file or folder using the custom detector you've defined.
+
+4. **Example**:
+
+   Let's use the template config provided above to search a file.
+
+   Assume you have a file `/tmp/data.txt` with the following content:
+
+   ```text
+   // this is a custom example
+   this file has some random text and maybe a secret
+   custom secret: token-JD44PMWmX4wBjqxE
+   // end of file
+   ```
+
+   In this file, the keyword `custom` exists, which will trigger the regex search. The string `token-JD44PMWmX4wBjqxE` matches the regex pattern, so it should be detected.
+
+   Run the following command:
+
+   ```bash
+   trufflehog filesystem /tmp --config=config.yaml
+   ```
+
+   The output should be similar to:
+
+   ```
+   🐷🔑🐷  TruffleHog. Unearth your secrets. 🐷🔑🐷
+
+   Found verified result 🐷🔑
+   Detector Type: CustomRegex
+   Decoder Type: PLAIN
+   Raw result: token-JD44PMWmX4wBjqxE
+   File: /tmp/data.txt
+   Line: 3
+   ```
+
+   The `Raw result` contains the matched string. `File` is the file name where secret was detected and `Line` is the exact line in the file where that was found.
+
+
+## Verification Server Examples
+Unless you run a verification server, secrets found by the custom regex detector will be unverified. Here is an example Python and Go implementation of a verification server for the above config.yaml file.
+
+### Python:
+
+```python
+import json
+from http.server import BaseHTTPRequestHandler, HTTPServer
+
+AUTH_HEADER = 'super secret authorization header'
+
+class Verifier(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(405)
+        self.end_headers()
+
+    def do_POST(self):
+        try:
+            if self.headers['Authorization'] != AUTH_HEADER:
+                self.send_response(401)
+                self.end_headers()
+                return
+
+            length = int(self.headers['Content-Length'])
+            request = json.loads(self.rfile.read(length))
+            self.log_message("%s", request)
+
+            if not validateTokens(request['CustomTokenDetector']['token']):
+                self.send_response(200)
+                self.end_headers()
+            else:
+                self.send_response(403)
+                self.end_headers()
+        except Exception:
+            self.send_response(400)
+            self.end_headers()
+
+def validateTokens(token):
+    return False  # Implement actual validation logic
+
+with HTTPServer(('', 8000), Verifier) as server:
+    try:
+        server.serve_forever()
+    except KeyboardInterrupt:
+        pass
+```
+
+### Go
+```go
+package main
```

Comment:
I think this only makes the concept of a "verification server" more confusing. The documentation needs to explain what it does and why you need it; examples in different languages doesn't provide clarity.

Reply:
The explanation is provided above in `verify` section where we link verification server examples. Shall I add more details there?

Reply:
I don't think there's any benefit in having multiple bare-bones examples.

Reply:
In my opinion, providing the example in the language TruffleHog is written in is beneficial.